network.go

```diff
@@ -57,6 +57,29 @@ func (network *_Network) _GetNetworkName() *NetworkName {
 	return network._ManagedNetwork._GetNetworkName()
 }
 
+func (network *_Network) _GetLedgerID() *LedgerID {
+	if network._ManagedNetwork._GetLedgerID() != nil {
+		return network._ManagedNetwork._GetLedgerID()
+	} else if network._ManagedNetwork._GetNetworkName() != nil {
+		temp, _ := LedgerIDFromNetworkName(*network._ManagedNetwork._GetNetworkName())
+		return temp
+	}
+
+	return &LedgerID{}
+}
+
+func (network *_Network) _SetLedgerID(id LedgerID) {
+	if id.IsMainnet() {//nolint
+		network._SetNetworkName(NetworkNameMainnet)
+	} else if id.IsTestnet() {
+		network._SetNetworkName(NetworkNameTestnet)
+	} else if id.IsPreviewnet() {
+		network._SetNetworkName(NetworkNamePreviewnet)
+	} else {
+		network._ManagedNetwork._SetLedgerID(id)
+	}
+}
+
 func (network *_Network) _SetNetworkName(net NetworkName) {
```

Comment:
Why not remove `NetworkName` from `Network` and use `LedgerID` instead? You can then convert between `LedgerID` and `NetworkName` using `LedgerID.ToNetworkName()`